modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/dao/JDBCResourceDAO.java

```diff
@@ -376,4 +381,237 @@ public boolean updateResource(String applicationKey, String resourceType, String
         return false;
     }
 
+    /**
+     * Add deployement config details
+     *
+     * @param deploymentConfig deployement config
+     */
+    public void addDeploymentConfig(DeploymentConfig deploymentConfig) throws AppFactoryException {
+        Connection connection = null;
+        PreparedStatement addDeploymentPreparedStatement = null;
+
+        try {
+            connection = AppFactoryDBUtil.getConnection();
+            addDeploymentPreparedStatement = connection.prepareStatement(SQLConstants.ADD_DEPLOYMENT);
+            addDeploymentPreparedStatement.setString(1, deploymentConfig.getDeploymentName());
+            addDeploymentPreparedStatement.setInt(2, deploymentConfig.getReplicas());
+            addDeploymentPreparedStatement.executeUpdate();
+
+            List<Container> containerList = deploymentConfig.getContainers();
+            addContainer(containerList, deploymentConfig.getDeploymentName(), connection);
+            if(addDeploymentPreparedStatement.getUpdateCount() > 0){
+                connection.commit();
+            }
+
+        } catch (SQLException e) {
+            try {
+                if (connection != null) {
+                    connection.rollback();
+                }
+            } catch (SQLException rollbackExcpetion) {
+                log.error("Error while rolling back the added application", rollbackExcpetion);
```

Comment:
This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/2468ef6d30141528a2e5f414e549da52ff652ae7